plugins/aws/resoto_plugin_aws/resource/redshift.py

```diff
@@ -0,0 +1,445 @@
+        "target_reserved_node_type": S("TargetReservedNodeType"),
+        "target_reserved_node_count": S("TargetReservedNodeCount"),
+    }
+    reserved_node_exchange_request_id: Optional[str] = field(default=None)
+    status: Optional[str] = field(default=None)
+    request_time: Optional[datetime] = field(default=None)
+    source_reserved_node_id: Optional[str] = field(default=None)
+    source_reserved_node_type: Optional[str] = field(default=None)
+    source_reserved_node_count: Optional[int] = field(default=None)
+    target_reserved_node_offering_id: Optional[str] = field(default=None)
+    target_reserved_node_type: Optional[str] = field(default=None)
+    target_reserved_node_count: Optional[int] = field(default=None)
+
+
+# AWS does not give us a cluster arn in the api, and we have to figure it out
+# from the cluster network arn and the cluster name.
+def _cluster_arn(ns_arn_and_cluster_id: Tuple[str, str]) -> str:
+    ns_arn = ns_arn_and_cluster_id[0]
+    cluster_id = ns_arn_and_cluster_id[1]
+    splitted = ns_arn.split(":")
+    return f"arn:aws:redshift:{splitted[3]}:{splitted[4]}:cluster:{cluster_id[1]}"
+
+
+@define(eq=False, slots=False)
+class AwsRedshiftCluster(AwsResource):
+    kind: ClassVar[str] = "aws_redshift_cluster"
+    api_spec: ClassVar[AwsApiSpec] = AwsApiSpec("redshift", "describe-clusters", "Clusters")
+
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "id": S("ClusterIdentifier"),
+        "tags": S("Tags", default=[]) >> ToDict(),
+        "name": S("ClusterIdentifier"),
+        "arn": ((S("ClusterNamespaceArn") >> F(lambda x: (x,))) + (S("ClusterIdentifier") >> F(lambda x: (x,))))
+        >> F(lambda cn: _cluster_arn(cn)),
+        "ctime": S("ClusterCreateTime"),
+        "mtime": K(None),
+        "atime": K(None),
+        "redshift_node_type": S("NodeType"),
+        "redshift_cluster_status": S("ClusterStatus"),
+        "redshift_cluster_availability_status": S("ClusterAvailabilityStatus"),
+        "redshift_modify_status": S("ModifyStatus"),
+        "redshift_master_username": S("MasterUsername"),
+        "redshift_db_name": S("DBName"),
+        "redshift_endpoint": S("Endpoint") >> Bend(AwsRedshiftEndpoint.mapping),
+        "redshift_automated_snapshot_retention_period": S("AutomatedSnapshotRetentionPeriod"),
+        "redshift_manual_snapshot_retention_period": S("ManualSnapshotRetentionPeriod"),
+        "redshift_cluster_security_groups": S("ClusterSecurityGroups", default=[])
+        >> ForallBend(AwsRedshiftClusterSecurityGroupMembership.mapping),
+        "redshift_vpc_security_groups": S("VpcSecurityGroups", default=[])
+        >> ForallBend(AwsRedshiftVpcSecurityGroupMembership.mapping),
+        "redshift_cluster_parameter_groups": S("ClusterParameterGroups", default=[])
+        >> ForallBend(AwsRedshiftClusterParameterGroupStatus.mapping),
+        "redshift_cluster_subnet_group_name": S("ClusterSubnetGroupName"),
```

Comment:
should this result in a `dependant_node` to the corresponding instance of `AwsEc2Subnet`?